### PR TITLE
Issue/13268 on item click

### PR DESCRIPTION
**Pull request by malinajirka**
Partially fixes #13268

This PR adds onItemClick listener to items on the Activity Type Filter screen. The DiffUtil is added so the item selector's animation doesn't get interrupted.

Merge instructions:
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/13469 is merged
2. Remove "Not ready for merge" label
3. Merge this PR

To test:
- Make sure the ActivityLogFiltersFeatureConfig is enabled
1. My site -> Activity Log
2. Click on "Activity Type" chip
3. Click on an unchecked row and notice it gets checked
4. Click on a checked row again and notice it gets unchecked

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt [hunk 1]
```diff
@@ -71,6 +73,19 @@ class ActivityLogTypeFilterViewModel @Inject constructor(
         }
     }
 
+    private fun onItemClicked(itemId: Int) {
+        (_uiState.value as? Content)?.let { it ->
+            val updatedList = it.items.map {
+                if (it is ListItemUiState.ActivityType && it.id == itemId) {
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt]: Np: We might want to declare the parameter explicitly instead of using `it` convention in a nested lambda.

File: WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt [hunk 2]
```diff
@@ -105,6 +107,33 @@ class ActivityLogTypeFilterViewModelTest : BaseUnitTest() {
         assertThat((viewModel.uiState.value as UiState.Content).items.size).isEqualTo(1 + activityTypeCount)
     }
 
+    @Test
+    fun `item is checked, when the user clicks on it`() = test {
+        val uiStates = init().uiStates
+        startVM()
+
+        ((uiStates.last() as Content).items[1] as ActivityType).let {
```
Comment (ashiagr) [inline WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt]: Np: IDE highlights let as "Redundant `let` call could be removed" 
Same in the following test case.

File: .idea/compiler.xml [hunk 3]
```diff
@@ -11,9 +11,22 @@
...
+      <module name="WordPress-Android-submodule.libs.editor.WordPressEditor" target="1.8" />
+      <module name="WordPress-Android-submodule.libs.image-editor.ImageEditor" target="1.8" />
+      <module name="WordPress-Android-submodule.libs.login.WordPressLoginFlow" target="1.8" />
+      <module name="WordPress-Android-submodule.libs.networking.WordPressNetworking" target="1.8" />
+      <module name="WordPress-Android-submodule.libs.stories-android.stories" target="1.8" />
+      <module name="WordPress-Android-submodule.libs.utils.WordPressUtils" target="1.8" />
+      <module name="WordPress-Android-submodule.mp4compose" target="1.8" />
+      <module name="WordPress-Android-submodule.photoeditor" target="1.8" />
+      <module name="WordPress-Android-submodule.WordPress" target="1.8" />
+      <module name="WordPress-Android-submodule.WordPressMocks" target="1.8" />
```
Comment (ashiagr) [inline .idea/compiler.xml]: Are these changes intentional?

**Review by ashiagr — APPROVED** [review]
Works as described 👍.
Just left 2 nitpicks. Feel free to merge after considering them.

File: .idea/compiler.xml (hunk 3 above)
Comment (malinajirka) [reply .idea/compiler.xml]: Nope, thanks for catching this ! ;)

Comment (malinajirka) [reply .idea/compiler.xml]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/13470/commits/9c48db7077c588a11ee8d1530bf43ce62648e613

File: WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt (hunk 2 above)
Comment (malinajirka) [reply WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/13470/commits/ff93099ecd3257bd1c478c7cfe6c43b1399939dd

File: WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt (hunk 1 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModel.kt]: Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/13470/commits/ff93099ecd3257bd1c478c7cfe6c43b1399939dd

**Comment by malinajirka** [comment]
Thanks for the review @ashiagr. I've fixed the issues.

**Comment by ashiagr** [comment]
LGTM 👍.
